frappe/utils/__init__.py

```diff
@@ -664,9 +664,57 @@ def is_markdown(text):
 		return not NON_MD_HTML_PATTERN.search(text)
 
 
+class cached_property:
```

Comment:
I'm keeping the name same so that we can switch without much hassle when we stop supporting 3.11 or lower.

Reply:
Also inheriting from functools.cached_property to support `isinstance(x, functools.cached_property)` calls outside framework and `__class_getitem__`